File: AtCorder/ABC004/b.py
#!/usr/bin/env python3
import sys

c1 = list(reversed(input().split()))
c2 = list(reversed(input().split()))
c3 = list(reversed(input().split()))
c4 = list(reversed(input().split()))


def echoeach(c):
    for i in range(0, 4):
        sys.stdout.write(c[i])
        if i != 3:
            sys.stdout.write(" ")
        else:
            sys.stdout.write("\n")


echoeach(c4)
echoeach(c3)
echoeach(c2)
echoeach(c1)

File: AtCorder/ABC004/b.nim
import  sequtils, strutils, algorithm, math

var
  c1 = reversed(readLine(stdin).split)
  c2 = reversed(readLine(stdin).split)
  c3 = reversed(readLine(stdin).split)
  c4 = reversed(readLine(stdin).split)

proc main(c: seq[string]) =
  for i in 0..3:
    stdout.write(c[i])
    if i != 3:
      stdout.write(' ')
    else:
      stdout.write("\n")

main(c4)
main(c3)
main(c2)
main(c1)
